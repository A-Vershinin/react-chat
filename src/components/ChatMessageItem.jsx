import React, { Fragment } from 'react';
import classNames from 'classnames';
import { withStyles } from 'material-ui/styles';
import Typography from 'material-ui/Typography';
import Paper from '@material-ui/core/Paper';

const styles = theme => ({
  message: {
    maxWidth: '70%',
    minWidth: '10%',
    padding: theme.spacing.unit,
    marginLeft: theme.spacing.unit * 2,
  },
  messageFromMe: {
    marginRight: theme.spacing.unit * 2,
    backgroundColor: '#e6dcff'
  },
});

const ChatMessageItem = props => {
  const {
    classes,
    message,
    isMessageFromMe,
    userAvatar
  } = props;

  return (
    <Fragment>
      {!isMessageFromMe && userAvatar}
      <Paper className={classNames(classes.message, isMessageFromMe && classes.messageFromMe)}>
        <Typography variant="caption">
          {message.sender}
        </Typography>
        <Typography variant="body1">
          {message.content}
        </Typography>
      </Paper>
      {isMessageFromMe && userAvatar}
    </Fragment>
  )
}

export default withStyles(styles)(ChatMessageItem);
