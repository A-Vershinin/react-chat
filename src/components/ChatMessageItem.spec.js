/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ChatMessageItem from './ChatMessageItem.jsx';

jest.mock('./Avatar', () => () => 'Avatar');
// Return a fixed timestamp when moment().fromNow() is called
jest.mock('moment', () => () => ({ fromNow: () => '2 days ago' }));

const mockProps = {
  classes: {
    message: 'some text',
    messageFromMe: 'some text',
    messageWrapper: 'some text',
    messageWrappperFromMe: 'some text',
    statusMessage: 'some text',
    statusMessageUser: 'some text',
  },
  statusMessage: true,
  sender: {
    firstName: 'firstName',
    lastName: 'lastName',
    username: 'username',
    _id: '12345',
  },
  activeUser: {
    isMember: true,
    isCreator: true,
    isChatMember: true,
    firstName: 'firstName',
    lastName: 'lastName',
    username: 'username',
  },
  content: 'some content',
  createdAt: '2018-03-16T10:53:23.200Z',
};

describe('<ChatMessageItem />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <ChatMessageItem {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <ChatMessageItem {...mockProps} />
              </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders when statusMessage false', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <ChatMessageItem {...mockProps} statusMessage />
              </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
