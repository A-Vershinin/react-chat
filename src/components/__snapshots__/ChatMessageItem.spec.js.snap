// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMessageItem /> renders correctly 1`] = `
<div
  className="ChatMessageItem-messageWrapper-33 some text"
>
  <p
    className="MuiTypography-root-37 MuiTypography-body1-46 ChatMessageItem-statusMessage-35 some text"
  >
    <span
      className="MuiTypography-root-37 MuiTypography-caption-47 ChatMessageItem-statusMessageUser-36 some text"
      style={
        Object {
          "color": "#ff5722",
        }
      }
    >
      firstName lastName
    </span>
    some content
    <span
      className="MuiTypography-root-37 MuiTypography-caption-47"
    >
      2 days ago
    </span>
  </p>
</div>
`;

exports[`<ChatMessageItem /> renders when ChatMessageItem disabled 1`] = `
<div
  className="ChatMessageItem-messageWrapper-33 some text"
>
  <p
    className="MuiTypography-root-37 MuiTypography-body1-46 ChatMessageItem-statusMessage-35 some text"
  >
    <span
      className="MuiTypography-root-37 MuiTypography-caption-47 ChatMessageItem-statusMessageUser-36 some text"
      style={
        Object {
          "color": "#ff5722",
        }
      }
    >
      firstName lastName
    </span>
    some content
    <span
      className="MuiTypography-root-37 MuiTypography-caption-47"
    >
      2 days ago
    </span>
  </p>
</div>
`;

exports[`<ChatMessageItem /> renders when statusMessage false 1`] = `
<div
  className="ChatMessageItem-messageWrapper-33 some text"
>
  <p
    className="MuiTypography-root-37 MuiTypography-body1-46 ChatMessageItem-statusMessage-35 some text"
  >
    <span
      className="MuiTypography-root-37 MuiTypography-caption-47 ChatMessageItem-statusMessageUser-36 some text"
      style={
        Object {
          "color": "#ff5722",
        }
      }
    >
      firstName lastName
    </span>
    some content
    <span
      className="MuiTypography-root-37 MuiTypography-caption-47"
    >
      2 days ago
    </span>
  </p>
</div>
`;
